packages/itwin/property-grid/src/components/Header.scss

```diff
@@ -4,12 +4,52 @@
 *--------------------------------------------------------------------------------------------*/
 
 .property-grid-react-panel-header {
-  display: flex;
-  align-items: center;
-  box-sizing: border-box;
-  height: calc(var(--iui-size-l) * 2);
+  display: grid;
+  grid-template-columns: auto auto;
+  grid-template-rows: auto;
```

Comment:
Maybe it would be possible to reduce css amount by using grid template? Something like this:
```css
.property-grid-react-panel-header {
  display: grid;
  grid-template-areas: "label tools";

  .header-title: {
    grid-area: label;
  }
  .header-tools: {
    grid-area: tools;
  }
  &.search-expanded {
    grid-template-areas: 
      "tools"
      "label";
  }
}
```